pkg/controller/petset/pet_set.go

```diff
@@ -226,12 +227,41 @@ func (psc *StatefulSetController) getPodsForStatefulSet(ps *apps.StatefulSet) ([
 	}
 	// TODO: Do we need to copy?
 	result := make([]*api.Pod, 0, len(pods))
+	// Check parent statefulset annotation to avoid fighting with other controllers, it's safer than using name prefix
 	for i := range pods {
-		result = append(result, &(*pods[i]))
+		if ps.UID == getStatefulSetUID(pods[i]) {
```

Comment:
The adoption was part of our upgrade plan from 1.4 to 1.5 so far, and we were going to document it. You're right, this would break the adoption of pods from PetSet in a StatefulSet.